### PR TITLE
Matt/map fixes

### PR DESCRIPTION
**Pull request by mattbarker016**

### CODE REVIEW

File: TCAT/Controllers/FavoritesTableViewController.swift [hunk 1]
```diff
@@ -29,17 +30,32 @@ class FavoritesTableViewController: UITableViewController, UISearchBarDelegate {
...
-        allStops = SearchTableViewManager.shared.getAllStops()
+        resultsSection = Section(type: .searchResults, items: [ItemType]())
+        
         tableView.register(BusStopCell.self, forCellReuseIdentifier: Constants.Cells.busIdentifier)
         tableView.register(SearchResultsCell.self, forCellReuseIdentifier: Constants.Cells.searchResultsIdentifier)
-
-        resultsSection = Section(type: .searchResults, items: [ItemType]())
+        tableView.emptyDataSetSource = self
+        tableView.emptyDataSetDelegate = self
+        // tableView.reloadEmptyDataSet()
```
Comment (myo3) [inline TCAT/Controllers/FavoritesTableViewController.swift]: delete commented out code if uneeded

File: TCAT/Controllers/FavoritesTableViewController.swift [hunk 2]
```diff
@@ -71,13 +87,12 @@ class FavoritesTableViewController: UITableViewController, UISearchBarDelegate {
...
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return resultsSection.items.isEmpty ? allStops.count : resultsSection.items.count
+        return resultsSection.items.count
     }
 
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        let searchBar = UISearchBar()
         searchBar.isTranslucent = true
-        searchBar.placeholder = Constants.Phrases.searchPlaceholder
+        searchBar.placeholder = Constants.Phrases.favoritesPlaceholder
```
Comment (myo3) [inline TCAT/Controllers/FavoritesTableViewController.swift:94]: nice use of constants

File: TCAT/Controllers/OnboardingViewController.swift [hunk 3]
```diff
@@ -73,7 +73,7 @@ class OnboardingViewController: PresentationController {
     /// Change the amount of messages in the view. The number of pages shown will equal the number of messages
     let detailLabelMessages = [
         
-        "Beautiful and simple end-to-end navigation app for TCAT. Made by AppDev.",
+        "A b eautiful and simple end-to-end navigation app for TCAT. Made by AppDev.",
```
Comment (myo3) [inline TCAT/Controllers/OnboardingViewController.swift]: extra space

File: TCAT/Controllers/RouteDetail+ContentViewController.swift [hunk 4]
```diff
@@ -27,9 +27,13 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
     var mapView: GMSMapView!
     var bounds = GMSCoordinateBounds()
 
-    var networkTimer: Timer? = nil
-    /// Number of seconds to wait before auto-refreshing network call, timed with live indicator
-    var networkRefreshRate: Double = LiveIndicator.INTERVAL * 1.0
+    var liveTrackingNetworkTimer: Timer? = nil
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+ContentViewController.swift]: no need for nil default value

File: TCAT/Controllers/RouteDetail+ContentViewController.swift [hunk 5]
```diff
@@ -27,9 +27,13 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
...
+    /// Number of seconds to wait before auto-refreshing live tracking network call call, timed with live indicator
+    var liveTrackingNetworkRefreshRate: Double = LiveIndicator.INTERVAL * 1.0
+    
+    var busDelayNetworkTimer: Timer? = nil
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+ContentViewController.swift]: same comment

File: TCAT/Controllers/RouteDetail+ContentViewController.swift [hunk 6]
```diff
@@ -233,14 +232,14 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
     
     /// Show banner if no other status banner exists; turns status bar light
     func showBanner(_ message: String, status: BannerStyle) {
-        if self.banner == nil {
+        // if self.banner == nil {
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+ContentViewController.swift]: delete commented out code

File: TCAT/Controllers/RouteDetail+ContentViewController.swift [hunk 7]
```diff
@@ -173,23 +162,33 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
...
     
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        networkTimer?.invalidate()
-        networkTimer = nil
-        UIApplication.shared.statusBarStyle = .default
-        banner?.dismiss()
-        banner = nil
+        liveTrackingNetworkTimer?.invalidate()
+        liveTrackingNetworkTimer = nil
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+ContentViewController.swift]: is it necessary to set it to nil? after you invalidate it its pretty much nil. just got to make sure before you use it again to set liveTrackingNetworkTimer to a new Timer

File: TCAT/Controllers/RouteDetail+ContentViewController.swift [hunk 8]
```diff
@@ -173,23 +162,33 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
...
+        busDelayNetworkTimer?.invalidate()
+        busDelayNetworkTimer = nil
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+ContentViewController.swift]: same comment

File: TCAT/Controllers/OnboardingViewController.swift (hunk 3 above)
Comment (mattbarker016) [reply TCAT/Controllers/OnboardingViewController.swift]: thanks!

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 9]
```diff
@@ -749,7 +758,7 @@ extension RouteOptionsViewController: UIViewControllerPreviewingDelegate {
             return nil
         }
         
-        routeDetailViewController.preferredContentSize = CGSize(width: 0.0, height: 0.0)
+        routeDetailViewController.preferredContentSize = .zero
```
Comment (myo3) [inline TCAT/Controllers/RouteOptionsViewController.swift:767]: noice

File: TCAT/Utilities/Extensions.swift [hunk 10]
```diff
@@ -175,7 +177,10 @@ extension JSON {
         dateFormatter.locale = Locale(identifier: "en_US_POSIX")
         dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZ"
         let date = dateFormatter.date(from: self.stringValue) ?? Date.distantPast
-        return date
+        
+        // Zero out seconds, floor the time
+        let dateComponents = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute], from: date)
+        return Calendar.current.date(from: dateComponents) ?? Date.distantPast
```
Comment (myo3) [inline TCAT/Utilities/Extensions.swift]: I created a `truncateSeconds(from date: Date)` in `Time.swift`. Its private but make it unprivate when you merge and use that for less repeated code

File: TCAT/Views/SummaryView.swift [hunk 11]
```diff
@@ -34,6 +34,9 @@ class SummaryView: UIView {
     /// The primary summary label
     fileprivate var mainLabel = UILabel()
     
+    /// The live indicator
+    fileprivate var liveIndicator = LiveIndicator(size: .small, color: .clear)
```
Comment (myo3) [inline TCAT/Views/SummaryView.swift:40]: why `fileprivate`? just make it `private` unless other classes in this SummaryView.swift file need to access these vars

Comment (mattbarker016) [reply TCAT/Views/SummaryView.swift:40]: I thought Swift changed syntax to fileprivate. I should look up the difference

File: TCAT/Controllers/RouteDetail+DrawerViewController.swift [hunk 12]
```diff
@@ -60,6 +64,24 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
         }
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        // Bus Delay Network Timer
+        if let timer = busDelayNetworkTimer {
+            timer.invalidate()
+        }
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+DrawerViewController.swift]: could just do the one-liner `busDelayNetworkTimer?.invalidate()`

File: TCAT/Controllers/RouteDetail+DrawerViewController.swift [hunk 13]
```diff
@@ -60,6 +64,24 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
...
+        busDelayNetworkTimer = Timer.scheduledTimer(timeInterval: busDelayNetworkRefreshRate, target: self, selector: #selector(getDelays),
+                                                    userInfo: nil, repeats: true)
+        busDelayNetworkTimer!.fire()
```
Comment (myo3) [inline TCAT/Controllers/RouteDetail+DrawerViewController.swift]: ik you set the timer just above but just to be safe use `?`

File: TCAT/Views/SummaryView.swift (hunk 11 above)
Comment (mattbarker016) [reply TCAT/Views/SummaryView.swift:40]: TIL https://stackoverflow.com/questions/46588704/what-is-the-difference-between-private-and-fileprivate-in-swift-4

**Pull request merged**